internal/services/elasticsan/elastic_san_resource_test.go

```diff
@@ -0,0 +1,244 @@
+package elasticsan_test
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/elasticsan/2023-01-01/elasticsans"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ElasticSANTestResource struct{}
+
+func TestAccElasticSAN_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccElasticSAN_zoneWithZRS(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config:      r.zoneWithZRS(data),
+			ExpectError: regexp.MustCompile("zones are not supported"),
+		},
+	})
+}
+
+func TestAccElasticSAN_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccElasticSAN_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("total_iops").Exists(),
+				check.That(data.ResourceName).Key("total_mbps").Exists(),
+				check.That(data.ResourceName).Key("total_size_in_tib").Exists(),
+				check.That(data.ResourceName).Key("total_volume_size_in_gib").Exists(),
+				check.That(data.ResourceName).Key("volume_group_count").Exists(),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccElasticSAN_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_elastic_san", "test")
+	r := ElasticSANTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
```

Comment:
Please also test scenarios of removing properties(For example, `extended_size_in_tib` is `optional`, at first, it is set to a value by `update`, then it is changed to `nil` by `basic`), this case can be updated to:
baisc -> update -> basic

Reply:
extended_size_in_tib cannot be reduced, add a validate and a test for this

```
 performing Update: unexpected status 400 with error: ElasticSanExtendedSizeUpdateNotAllowed: Requested size update 0 cannot be performed as new extended size should be greater than existing extended size

          with azurerm_elastic_san.test,
          on terraform_plugin_test.tf line 38, in resource "azurerm_elastic_san" "test":
          38: resource "azurerm_elastic_san" "test" {
```